ivr/src/main/java/org/motechproject/ivr/service/impl/OutboundCallServiceImpl.java

```diff
@@ -130,6 +151,32 @@ public void initiateCall(String configName, Map<String, String> parameters) {
             addCallDetailRecord(CallDetailRecord.CALL_FAILED, config, params, motechCallId);
             throw new CallInitiationException(message, e);
         }
+    }
+
+    private void generateCallFile(Config config, Map<String, String> params, Map<String, String> completeParams, String motechCallId) throws URISyntaxException, IOException {
+        String myUri = mergeUriAndRemoveParams(config.getOutgoingCallUriTemplate(), completeParams);
+
+        Path path = Paths.get(new URI(myUri));
+        Path tempDirectory = Files.createTempDirectory(path.getParent(), "motech");
```

Comment:
according to Asterisk wiki (https://wiki.asterisk.org/wiki/display/AST/Asterisk+Call+Files) it is recommended to create a file in another directory on the same filesystem